e2e/test/scenarios/actions/reproductions/31587-bottom-corner-radius-action-buttons.cy.spec.js

```diff
@@ -0,0 +1,70 @@
+import {
+  editDashboard,
+  restore,
+  saveDashboard,
+  setActionsEnabledForDB,
+  visitDashboard,
+} from "e2e/support/helpers";
+import { SAMPLE_DB_ID } from "e2e/support/cypress_data";
+
+const sizes = [
+  [768, 800],
+  [1024, 800],
+  [1440, 800],
+];
+describe("metabase#31587", () => {
+  sizes.forEach(([width, height]) => {
+    beforeEach(() => {
+      restore();
```

Comment:
Let's imagine there's an extra line here:

```suggestion
      restore();
      console.log("how many times will this print?");
```

I believe the intention was that this will be printed 6 times (2 `it` times 3 viewports).

In fact it will be printed 18 times.

I suggest to replace the `context` call with `describe` call and wrap the entire `forEach` callback with it.

Reply:
Ah, great catch. From what I saw on the Cypress dashboard it really seemed like it just ran 6 times
<img width="335" alt="image" src="https://github.com/metabase/metabase/assets/25306947/4c5afb01-d5da-488a-ae57-8e069595e24b">